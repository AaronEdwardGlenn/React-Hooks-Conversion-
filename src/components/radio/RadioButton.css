.RadioButton {
    label {
    border-radius: 10%;
     padding: .5rem;

     &:hover {
         background-color: blueviolet;
     }
    }
   

    input:checked + label {
        background-color: green;
        color: yellow
    }
}
