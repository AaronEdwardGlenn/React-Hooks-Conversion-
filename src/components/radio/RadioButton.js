import React from 'react';
import PropTypes from 'prop-types';
import styles from './RadioButton.css';

const RadioButton = ({ name, label, value, handleChange }) => (

  <div className={styles.RadioButton}>
    <input type="radio" name={name} label={label} value={value} onChange={handleChange} />
    <label htmlFor={value}>{label}</label>
  </div>
);

RadioButton.propTypes = {
  name: 
    PropTypes.string.isRequired,
  label: 
    PropTypes.string.isRequired,
  value: 
    PropTypes.string.isRequired,
  selected: 
  PropTypes.string.isRequired,
  handleChange: 
  PropTypes.func.isRequired,
};

export default RadioButton; 
