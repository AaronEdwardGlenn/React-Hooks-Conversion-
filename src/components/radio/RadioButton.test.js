import React from 'react';
import { shallow } from 'enzyme';
import RadioButton from './RadioButton';

describe('radiobuttons component', () => {
  it('matches the snap', () => {
    const wrapper = shallow(<RadioButton name="rad name" label = "rad lable" value = "rad value" onChange = {() => {}} />);
    expect(wrapper).toMatchSnapshot(); 
  });
  
});
