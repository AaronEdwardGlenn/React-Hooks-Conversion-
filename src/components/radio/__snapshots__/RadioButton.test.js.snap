// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radiobuttons component matches the snap 1`] = `
<div
  className="RadioButton"
>
  <input
    label="rad lable"
    name="rad name"
    type="radio"
    value="rad value"
  />
  <label
    htmlFor="rad value"
  >
    rad lable
  </label>
</div>
`;
